.github/workflows/reusable-mypy.yml

```diff
@@ -0,0 +1,21 @@
+on:
+  workflow_call:
+
+jobs:
+  check-with-mypy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
```

Comment:
We should probably find a way to make sure that `mypy` uses the correct Python version for the specific repo. Maybe the easiest way would be to take the version as an input to the reusable action, and then pass the version to `mypy` via the [`--python-version`](https://mypy.readthedocs.io/en/stable/command_line.html#cmdoption-mypy-python-version) option?

Reply:
Unless there's a way to make `mypy` use the version from `pyproject.toml` like `ruff` does?